### PR TITLE
www-misc/buku: Bump to 4.4

### PR DESCRIPTION
**Pull request by ajakk**
Package-Manager: Portage-3.0.0, Repoman-2.3.23
Signed-off-by: John Helmert III <jchelmert3@posteo.net>

### CODE REVIEW

File: www-misc/buku/buku-4.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
```
Comment (thesamesam) [inline www-misc/buku/buku-4.4.ebuild:6]: What's in the way of Python 3.9?

File: www-misc/buku/buku-4.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	dev-python/cryptography[${PYTHON_USEDEP}]
+	>=dev-python/html5lib-1.0.1[${PYTHON_USEDEP}]
+	dev-python/urllib3[${PYTHON_USEDEP}]
+"
+
+DEPEND="${RDEPEND}
+	test? (
+		dev-python/attrs[${PYTHON_USEDEP}]
+		>=dev-python/click-7.0[${PYTHON_USEDEP}]
+		dev-python/flake8[${PYTHON_USEDEP}]
```
Comment (thesamesam) [inline www-misc/buku/buku-4.4.ebuild:32]: Remove flake8, pylint if [possible](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#undesirable-test-dependencies)

File: www-misc/buku/buku-4.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+DESCRIPTION="Powerful command-line bookmark manager"
+HOMEPAGE="https://github.com/jarun/buku"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="test"
```
Comment (thesamesam) [inline www-misc/buku/buku-4.4.ebuild:19]: Why?

File: www-misc/buku/buku-4.4.ebuild (hunk 1 above)
Comment (stkw0) [reply www-misc/buku/buku-4.4.ebuild:6]: upstream doesn't support python3.9 for now

Comment (thesamesam) [reply www-misc/buku/buku-4.4.ebuild:6]: But does it work? A lot of upstreams won't specifically mention it for now, and users enabling it know the risks.

File: www-misc/buku/buku-4.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	doins auto-completion/zsh/_*
+
+	newbashcomp auto-completion/bash/buku-completion.bash "${PN}"
+
+	doman buku.1
+}
+
+python_test() {
+	# Explicitly enumerate tests to avoid tests that deal with bukuserver
+	py.test -v "tests/test_"{BukuCrypt,bukuDb,buku,ExtendedArgumentParser,import_firefox_json}".py" || die
```
Comment (thesamesam) [inline www-misc/buku/buku-4.4.ebuild:63]: || die "Tests failed with ${EPYTHON}", but I know they're restricted at the moment.

File: www-misc/buku/buku-4.4.ebuild (hunk 1 above)
Comment (stkw0) [reply www-misc/buku/buku-4.4.ebuild:6]: I was told that the gentoo policy is to follow what upstreams supports (maybe some dev can correct me if I'm wrong). Also python3.9 is in beta.
I suppose we can add python3.9 but for now I don't think we have any reason to do it.

Comment (ajakk) [reply www-misc/buku/buku-4.4.ebuild:6]: There is a broken test with Python 3.9, should probably stay disabled for now, though I've [reported it upstream](https://github.com/jarun/buku/issues/464).

File: www-misc/buku/buku-4.4.ebuild (hunk 3 above)
Comment (ajakk) [reply www-misc/buku/buku-4.4.ebuild]: We now skip tests that are broken in the Gentoo environment.

**Pull request closed without merging**